GitUI/CommandsDialogs/FormDeleteRemoteBranch.cs

```diff
@@ -45,33 +46,55 @@ protected override void OnRuntimeLoad(EventArgs e)
 
         protected override void OnShown(EventArgs e)
         {
-            RecalculateSizeConstraints();
+            CheckDeleteTrackingAllowed();
             base.OnShown(e);
             Branches.Focus();
         }
 
-        private void RecalculateSizeConstraints()
+        private List<IGitRef> GetSelectedRemotRefs() => Branches.GetSelectedBranches().ToList();
+        private void CheckDeleteTrackingAllowed()
         {
-            SuspendLayout();
-            MinimumSize = MaximumSize = Size.Empty;
+            string[] localTracking = GetTrackingReferenceOfRemoteRefs(GetSelectedRemotRefs());
+            bool localTrackingBranchesExists = localTracking.Any();
+            const int maxDisplayed = 8;
 
-            int height = ControlsPanel.Height + MainPanel.Padding.Top + MainPanel.Padding.Bottom
-                       + tlpnlMain.Height + tlpnlMain.Margin.Top + tlpnlMain.Margin.Bottom + DpiUtil.Scale(42);
-
-            MinimumSize = new Size(tlpnlMain.PreferredSize.Width + DpiUtil.Scale(70), height);
-            MaximumSize = new Size(Screen.PrimaryScreen.Bounds.Width, height);
-            Size = new Size(Width, height);
-            ResumeLayout();
+            if (!localTrackingBranchesExists)
+            {
+                DeleteLocalTrackingBranch.Checked = false;
+                DeleteLocalTrackingBranch.Enabled = false;
+                _NO_TRANSLATE_labelLocalTrackingBranches.Text = string.Empty;
+            }
+            else
+            {
+                DeleteLocalTrackingBranch.Enabled = true;
+                _NO_TRANSLATE_labelLocalTrackingBranches.Text = _toDeleteCandidates.Text
+                    + $"{Environment.NewLine} - "
+                    + string.Join($"{Environment.NewLine} - ", localTracking.Take(maxDisplayed));
+                if (localTracking.Length > maxDisplayed)
+                {
+                    _NO_TRANSLATE_labelLocalTrackingBranches.Text += $"{Environment.NewLine}{Environment.NewLine} "
+                        + string.Format(_andMore.Text, localTracking.Length - maxDisplayed);
```

Comment:
Prefer string interpolation (or at least stringbuilder) as it has better perf characteristics.